### ISSUE TITLE
Double dice with Better Rolls and Dice so Nice.

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#708]
In GitLab by @gantonaci on Feb 22, 2022, 08:39

With Better Rolls and Dice so Nice active, the damage dice from a BonusDamage macro is rolling twice (the damage is considered only once). Macro used is simply:

```return { damageRoll: `1d6[fire]` , flavor: "Fire Damage" }```

Example: https://youtu.be/8sTT9mvE9IQ

Video is before I deactivated modules to find the problem, but the issue is exactly the same.

Tested as GM and as a player, problem occurs for both.

Console log:
```
itemhandling.js:96 Advantage/Disadvantage sources:  []
doAttackRoll @ itemhandling.js:96
await in doAttackRoll (async)
🎁call_wrapper @ libWrapper-wrapper.js:620
🎁CONFIG.Item.documentClass.prototype.rollAttack#0 @ libWrapper-wrapper.js:189
getAttackRoll @ utils.js:590
constructAttackRoll @ fields.js:171
constructModelsFromField @ fields.js:438
_processField @ custom-roll.js:976
await in _processField (async)
roll @ custom-roll.js:701
await in roll (async)
toMessage @ custom-roll.js:824
itemRoll @ index.js:63
🎁call_wrapper @ libWrapper-wrapper.js:620
doItemRoll @ itemhandling.js:487
🎁call_wrapper @ libWrapper-wrapper.js:620
🎁CONFIG.Item.documentClass.prototype.roll#0 @ libWrapper-wrapper.js:189
rollItemMacro @ macros.js:58
eval @ VM22831:5
callScriptMacroFunction @ Macros.js:172
renderMacro @ Macros.js:188
executeMacro @ Macros.js:209
_onClickMacro @ foundry.js:52259
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
backend.mjs:306 Foundry VTT | Created ChatMessage with id [XLBgCYoX1AfD67AB]
midi-qol.js:32 midi-qol |  Hidden/Invisibility removed for Mara due to attack
midi-qol.js:32 midi-qol |  Mara Rolled a 22 to hit Gnoll(64)'s AC of 1 hitting
midi-qol.js:32 midi-qol |  Gnoll(64) 306 takes 12 reduced from 12 Temp HP 0 HP 294 
custom-roll.js:272 BetterRolls | Card loaded existing item data Eldritch Blast
backend.mjs:306 Foundry VTT | Created ChatMessage with id [6jmDX9xMCTVyw7ak]
chat-message.js:80 BetterRolls5e | Retrieved existing card
backend.mjs:306 Foundry VTT | Created ChatMessage with id [9XzXjxPTa9IyEWzp]
midi-qol.js:32 midi-qol |  createReverseDamageCard elapsed: 151
chat-message.js:55 BetterRolls5e | Hover Buttons Initialized
backend.mjs:306 Foundry VTT | Created ChatMessage with id [Ha5itXBEGMfjjQu6]
backend.mjs:306 Foundry VTT | Created ChatMessage with id [e8qSf7BphLWnoEsq]
chat-message.js:80 BetterRolls5e | Retrieved existing card
```

[fvtt-midi-qol-settings__3_.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/89141e5b11fb28901244d4671460bc6844531d410c99c723bb8ca3b6746b32a6/fvtt-midi-qol-settings__3_.json)

**Comment by tposney** [comment]
Thanks for the video it makes it much clearer.
I'll have a look into this.

**Comment by tposney** [comment]
Fix will go out in 0.9.23
